app/demo-geant-integration/SensitiveDetector.cc

```diff
@@ -41,9 +54,29 @@ G4bool SensitiveDetector::ProcessHits(G4Step* step, G4TouchableHistory*)
         return false;
     }
 
-    auto time = step->GetPreStepPoint()->GetGlobalTime();
+    // Fill the hit data for this step
+    auto         touchable = step->GetPreStepPoint()->GetTouchable();
+    unsigned int id        = touchable->GetVolume()->GetCopyNo();
+    HitData      data{id,
+                 edep,
+                 step->GetPreStepPoint()->GetGlobalTime(),
+                 touchable->GetTranslation()};
```

Comment:
Shouldn't you create the `HitData` only after searching for existing hits?

Reply:
You are correct.  Actually I remove the reduction by the volume id as we may keep all hits inside the same volume with different time and position information.  Anyway, we will redefine sensitive hits and their store as close to CMS-like in the second pass.